packages/cli/src/modules/mcp/tools/get-workflow-details.tool.ts

```diff
@@ -0,0 +1,84 @@
+import type { User, WorkflowEntity } from '@n8n/db';
+import { OperationalError } from 'n8n-workflow';
+import z from 'zod';
+
+import type { ToolDefinition } from '../mcp.types';
+import { getWebhookDetails } from './utils';
+
+import type { CredentialsService } from '@/credentials/credentials.service';
+import type { WorkflowFinderService } from '@/workflows/workflow-finder.service';
+
+const inputSchema = {
+	workflowId: z.string().describe('The ID of the workflow to retrieve'),
+} satisfies z.ZodRawShape;
+
+export const createWorkflowDetailsTool = (
+	user: User,
+	baseWebhookUrl: string,
+	workflowFinderService: WorkflowFinderService,
+	credentialsService: CredentialsService,
+): ToolDefinition<typeof inputSchema> => {
+	return {
+		name: 'get_workflow_details',
+		config: {
+			description: 'Get detailed information about a specific workflow including trigger details',
+			inputSchema,
+		},
+		handler: async ({ workflowId }) => {
+			const payload = await getWorkflowDetails(
+				user,
+				baseWebhookUrl,
+				workflowFinderService,
+				credentialsService,
+				{ workflowId },
+			);
+
+			return {
+				content: [{ type: 'text', text: JSON.stringify(payload) }],
+			};
+		},
+	};
+};
+
+export async function getWorkflowDetails(
+	user: User,
+	baseWebhookUrl: string,
+	workflowFinderService: WorkflowFinderService,
+	credentialsService: CredentialsService,
+	{ workflowId }: { workflowId: string },
+): Promise<{ workflow: WorkflowEntity; triggerInfo: string }> {
+	const workflow = await workflowFinderService.findWorkflowForUser(workflowId, user, [
+		'workflow:read',
+	]);
+	if (!workflow || workflow.isArchived || !workflow.settings?.availableInMCP) {
+		throw new OperationalError('Workflow not found');
```

Comment:
Up for debate, but I think `UserError` might be more appropriate. It's not a temporary condition, i.e. something that goes away over time